iP/src/duke/Duke.java

```diff
@@ -0,0 +1,172 @@
+     */
+    public void user_input_handler(String user_input, boolean loading) throws DukeException, IOException {
+        String instructionType = Parser.parseInstruction(user_input);
+
+        if (instructionType.equals("bye")) {
+            // quit
+            ui.showGoodBye();
+            System.exit(0);
+
+        } else if (instructionType.equals("list")) {
+            // list task
+            ui.showListOfTask(tasks.taskList);
+
+        } else if (instructionType.equals("done")) {
+            // mark done
+            // parse instruction
+            int index = Parser.parseMarkDoneInstr(user_input);
+            // execute
+            Task chosenTask = tasks.getTask(index);
+            chosenTask.markAsDone();
+            if (!loading){
+                ui.showMarkedDoneTask(chosenTask);
+            }
+
+        } else if (instructionType.equals("delete")) {
+            // delete task
+            // parse instruction
+            int index = Parser.parseDeleteInstr(user_input);
+            // execute
+            Task chosenTask = tasks.getTask(index);
+            tasks.deleteTask(index);
+            if (!loading){
+                ui.showDeletedTask(chosenTask, tasks.taskList);
+            }
+
+        } else if (instructionType.equals("find")) {
+            // find task
+            // parse instruction
+            String keyword = Parser.parseFindInstr(user_input);
+            // execute
+            ArrayList<Task> foundTasks = tasks.find(keyword);
+            if (!loading){
+                ui.showFoundTask(foundTasks);
+            }
+
+        } else if (instructionType.equals("todo")) {
+            // make todo
+            try {
+                // parse instruction
+                String description = Parser.parseAddTodoInstr(user_input);
+                // execute
+                Task todo = new Todo(description);
+                tasks.addTask(todo);
+                if (!loading){
+                    ui.showAddedTask(todo, tasks.taskList);
+                }
+            } catch (DukeException e) {
+                ui.showError(e.getMessage());
+            }
+
+        } else if (instructionType.equals("deadline")) {
+            // make deadline
+            try {
+                // parse instruction
+                HashMap<String, Object> parsedData = Parser.parseAddDeadlineInstr(user_input);
+                String description = (String) parsedData.get("description");
+                LocalDate l_time = (LocalDate) parsedData.get("time");
+                // execute
+                Task deadline = new Deadline(description, l_time);
+                tasks.addTask(deadline);
+                if (!loading) {
+                    ui.showAddedTask(deadline, tasks.taskList);
+                }
+            } catch (DukeException e) {
+                ui.showError(e.getMessage());
+            }
+
+        } else if (instructionType.equals("event")){
```

Comment:
You might have missed a white space here

Reply:
I second that